slider/curve.py

```diff
@@ -179,57 +203,48 @@ def __call__(self, t):
         raise NotImplementedError('catmull positions not supported yet')
 
 
-def get_center(p1, p2, p3):
+def get_center(a, b, c):
     """Returns the Position of the center of the circle described by the 3
     points
 
     Parameters
     ----------
-    p1, p2, p3 : Position
+    a, b, c : Position
         The three positions.
 
     Returns
     -------
     center : Position
         The center of the three points.
+
+    Notes
+    -----
+    This uses the same algorithm as osu!
+    https://github.com/ppy/osu/blob/7fbbe74b65e7e399072c198604e9db09fb729626/osu.Game/Rulesets/Objects/CircularArcApproximator.cs#L23  # noqa
     """
-    if p2.x == p1.x:
-        # normal t calc won't work
-        t = (p3.y - p1.y) / (2 * (p3.x - p2.x))
+    a, b, c = np.array([a, b, c])
 
-        return Position(
-            0.5 * (p2.x + p3.x) + t * (p3.y - p2.y),
-            0.5 * (p2.y + p3.y) - t * (p3.x - p2.x),
-        )
-    elif p3.y == p2.y:
-        t = (p3.x - p1.x) / (2 * (p2.y - p1.y))
-    else:
-        t = (
-            (
-                (-(p1.y - p3.y) / (2 * (p2.x - p1.x))) -
-                (
-                    ((p3.x - p2.x) * (p1.x - p3.x)) /
-                    (2 * (p2.x - p1.x) * (p3.y - p2.y))
-                )
-            ) *
-            (
-                ((p2.x - p1.x) * (p3.y - p2.y)) /
-                (
-                    ((p3.x - p2.x) * (p2.y - p1.y)) -
-                    ((p2.x - p1.x) * (p3.y - p2.y))
-                )
-            )
-        )
+    a_squared = np.sum(np.square(b - c))
+    b_squared = np.sum(np.square(a - c))
+    c_squared = np.sum(np.square(a - b))
 
-    return Position(
-        0.5 * (p1.x + p2.x) + t * (p2.y - p1.y),
-        0.5 * (p1.y + p2.y) - t * (p2.x - p1.x),
-    )
+    if np.isclose([a_squared, b_squared, c_squared], 0).any():
+        raise ValueError()
```

Comment:
I agree. These checks guard against collinear points, I will put that in the error message with the coordinates.